tests/unit/test_alpaca_websocket.py

```
@@ -496,110 +496,84 @@ def test_unsubscribe_when_not_connected(self):
+                assert (
+                    actual_delay == expected_delay
+                ), f"Attempt {attempt}: {actual_delay} != {expected_delay}"
+
+    def test_reconnect_max_attempts_check(self):
+        """Test that max reconnection attempts is enforced."""
         with (
             patch("core.data.alpaca_client.TradingClient"),
             patch("core.data.alpaca_client.StockHistoricalDataClient"),
-            patch("core.data.alpaca_client.StockDataStream") as mock_stream,
         ):
             client = AlpacaClient(api_key="test_key", secret_key="test_secret")
```

Ruff (0.14.8)

Possible hardcoded password assigned to argument: "secret_key"

(S106)